### PR TITLE
Vendor onboarding

### PR DESCRIPTION
**Pull request by sztaylorakgov**
Finished rough draft of vendor onboarding checklist.  Notably, this isn't really a `checklist` yet.  We can fix that if needed, but there are some more fundamental questions to hash through in my view.  I'll make some specific comments and request some feedback.

### CODE REVIEW

File: vendor-onboarding-checklist.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+# Roles
+
+This section identifies the anticipated vendor roles that will be
+engaged with the
+project.
+
+| **Role**                 | **Description**                                                                             |
+| ------------------------ | ------------------------------------------------------------------------------------------- |
+| Project manager          | Manage the project                                                                          |
```
Comment (sztaylorakgov) [inline vendor-onboarding-checklist.md]: @nfaulknerAK Is the project manager and the scrum master two different vendor resources in your mind?  It seems a little redundant to me.

File: vendor-onboarding-checklist.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+<li><p>CRU docs on SharePoint</p></li>
+<li><p>Read and update work-items and PR comments in VSTS</p></li>
+<li><p>Read and update Trello</p></li>
+<li><p>Read and post messages to Slack</p></li>
+<li><p>Read GitHub repositories and issues</p></li>
+<li><p>Participate in Mural sprint retrospectives</p></li>
+</ul></td>
+</tr>
+<tr class="even">
+<td>Technical lead</td>
```
Comment (sztaylorakgov) [inline vendor-onboarding-checklist.md]: @ctro Can you take a look at the list of resources the tech lead and dev's will need access to and provide some feedback?

File: vendor-onboarding-checklist.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+This section outlines procedures to grant
+access.
+
+| **Resource**                   | **Procedure**                            |
+| ------------------------------ | ---------------------------------------- |
+| State of Alaska LDAP account   |                                          |
+| State of Alaska VPN            |                                          |
+| DHSS domain account            |                                          |
+| DHSS SharePoint                | Request via Alaska EIS-R PMO             |
```
Comment (sztaylorakgov) [inline vendor-onboarding-checklist.md]: @ctro @nfaulknerAK @mheadd I think this level of detail could be fine for this document.  We will need more specific procedures and forms on the backend.

File: vendor-onboarding-checklist.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+| Visual Studio Team Services    | Request via Alaska Business Applications |
+| Trello                         | ?                                        |
+| Slack                          | ?                                        |
+| Alaska DHSS GitHub             | Request via Alaska Business Applications |
+| Mural                          | ?                                        |
+| Alaska Development Environment | Request via Alaska Business Applications |
+
+# Training resources
```
Comment (sztaylorakgov) [inline vendor-onboarding-checklist.md]: This section is, technically, beyond the scope of Trello 195.  However, I do think we need to baseline expectations about how vendor staff will learn to use the resources they have access to.

File: vendor-onboarding-checklist.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+<li><p>Participate in Mural sprint retrospectives</p></li>
+</ul></td>
+</tr>
+</tbody>
+</table>
+
+# Access procedures
+
+This section outlines procedures to grant
+access.
```
Comment (sztaylorakgov) [inline vendor-onboarding-checklist.md]: We need to add some info here to clarify that access is sponsored and that the DPA PMO sponsors and manages access requests on behalf of vendors.

File: vendor-onboarding-checklist.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+| Project manager          | Manage the project                                                                          |
+| Technical lead           | Lead the technical work                                                                     |
+| Developers               | Develop client and server application components using CSS, HTML5, Javascript and .NET Core |
+| Researcher               |                                                                                             |
+| User experience designer | Design the user experience                                                                  |
+| Visual designer          |                                                                                             |
+| Scrum Master             | Lead internal daily scrum sessions                                                          |
+
+# Tools and resources
+
```
Comment (sztaylorakgov) [inline vendor-onboarding-checklist.md]: Review team in the DevSecOps road show weekly recommended we pull out common items to help reduce duplication and clarify what unique access is granted to different roles.

Comment (sztaylorakgov) [reply vendor-onboarding-checklist.md]: Done.

File: vendor-onboarding-checklist.md (hunk 5 above)
Comment (sztaylorakgov) [reply vendor-onboarding-checklist.md]: Done.

File: vendor-onboarding-checklist.md (hunk 1 above)
Comment (nfaulknerAK) [reply vendor-onboarding-checklist.md]: I don't think they are two different resources. I agree that would be redundant.

Comment (sztaylorakgov) [reply vendor-onboarding-checklist.md]: Per slack convo, combining PM and Scrum master roles in vendor-onboarding-checklist.

File: vendor-onboarding-checklist.md (hunk 2 above)
Comment (ctro) [reply vendor-onboarding-checklist.md]: For the "Able to" section should we include specific abilities in VSTS? 

- abilities in the "Code" section; they can probably have pretty full reign except on `staging` and `master` branches.
- abilities in the "Build" section; can they edit build definitions? 
- abilities in the "Release" section; can they queue new releases?  edit release definitions?

That kinda takes me to Azure access/abilities. Should we add Azure? We plan on having 3 AppService Slots to begin with: prod, staging, dev.  can these people manually deploy to the development slot?  can they access the Kudu dashboard for Azure dev slot?

File: vendor-onboarding-checklist.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+<p>Trello</p>
+<p>Slack</p>
+<p>Alaska DHSS GitHub</p>
+<p>Mural</p></td>
+<td><ul>
+<li><p>Access Alaska WAN</p></li>
+<li><p>Access DHSS domain secured resources</p></li>
+<li><p>CRU docs on SharePoint</p></li>
+<li><p>Read and update work-items and PR comments in VSTS</p></li>
+<li><p>Read and update Trello</p></li>
```
Comment (ctro) [inline vendor-onboarding-checklist.md]: I think it's good for us to leave in tools like Trello that we currently use but might not in the future.

File: vendor-onboarding-checklist.md (hunk 3 above)
Comment (ctro) [reply vendor-onboarding-checklist.md]: :+1: Agree. No need to get into the nitty-gritty.  I think just providing the entry-point is best.

File: vendor-onboarding-checklist.md (hunk 6 above)
Comment (ctro) [reply vendor-onboarding-checklist.md]: The "Tools and Resources" list is also pretty wet.
As in the opposite of DRY.
(Don't Repeat Yourself) :)

File: vendor-onboarding-checklist.md (hunk 3 above)
Comment (sztaylorakgov) [reply vendor-onboarding-checklist.md]: Made some minor mods to the language from header of "Procedure" to "Requested and granted via".

File: vendor-onboarding-checklist.md (hunk 2 above)
Comment (sztaylorakgov) [reply vendor-onboarding-checklist.md]: We should chat about this stuff for sure.  I don't know what "full reign" in the Code section means.  What I imagine is that they can push new branches, create PRs.  What else would they need to do?  E.g., I wouldn't want them to change branch policies.  

I'm starting to think that I need to reformat the doc and pull it out of table format.  My tendency as you say things like "full reign" or "include specific abilities in VSTS" is to start listing out the sections (e.g., "Code") and subsections (e.g., "Files", "Commits", "Pushes", "Branches", "Tags", "Pull requests") and then identify what they can do.

On the other hand, I think what we're running into is that we don't have the roles defined yet.  So, there's a more detailed exercise of going through VSTS and thinking about "Does a tech lead need to do X?" or "Does a dev need to do X".  And this feels like a necessary exercise, but a totally different document from the "vendor-onboarding-checklist".  Thoughts?

**Comment by ctro** [comment]
@sztaylorakgov I somehow missed the thread in here about specifying controls for different sections in VSTS -- but wanted to say that you were right: I was talking about vendors not being able to change `staging` and `master` branch configurations. And similar questions for Builds, and also for accessing Azure AppService environments.

I definitely don't think we need to go off the deep end here, maybe there's an easy way to get the point across w/o being too specific.  Like "vendors will have full access to development environments and tools, but restricted access to staging and production environments and tools" ?

File: vendor-onboarding-checklist.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+| Technical lead                 | Lead the technical work                                                                                                                                        |
+| Developers                     | Develop client and server application components using CSS, HTML5, Javascript and .NET Core                                                                    |
+| Researcher                     |                                                                                                                                                                |
+| User experience designer       | Design the user experience to achieve good continuity between new and legacy application and usability that yields the intended value of the product increment |
+| Visual designer                |                                                                                                                                                                |
+
+# Tools and resources
```
Comment (dvvarela) [inline vendor-onboarding-checklist.md]: I think that this table should be broken down into 2 different ones. The first a list of Tools/Resources and the purpose. The second table, a matrix of Tool/Resources to Roles.

tools-resources-purpose
tools-resources-roles

File: vendor-onboarding-checklist.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+| State of Alaska VPN            | DHSS IT Help Desk             |
+| DHSS domain account            | DHSS IT Help Desk             |
+| DHSS SharePoint                | DHSS EIS-R PMO                |
+| Visual Studio Team Services    | DHSS Business Applications    |
+| Trello                         | TTS/18F                       |
+| Slack                          | TTS/18F                       |
+| Alaska DHSS GitHub             | DHSS Business Applications    |
+| Mural                          | TTS/18F                       |
+| Alaska Development Environment | DHSS Business Applications    |
+
```
Comment (dvvarela) [inline vendor-onboarding-checklist.md]: We may want to consider adding a section with required forms and supporting documentation to be submitted.

Non-State employees are required to submit the Security Agreement (located on internal DPAWeb), along with supporting documentation (verifying identity) and HIPAA training certificate.

Comment (randyhart) [reply vendor-onboarding-checklist.md]: I agree - getting as much of the required forms and paperwork into one place will help the winning vendor hit  the ground running.

File: vendor-onboarding-checklist.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+This section identifies the tools and resources that vendors will need
+access to work with and what type of access they will need.
+
+<table>
+<thead>
+<tr class="header">
+<th><strong>Role</strong></th>
+<th><strong>Role-specific tools and resources</strong></th>
+<th><strong>Able to</strong></th>
```
Comment (ctro) [inline vendor-onboarding-checklist.md]: The "Able to" section is looking good to me: hits all the big targets w/o getting into the weeds.

File: vendor-onboarding-checklist.md (hunk 8 above)
Comment (sztaylorakgov) [reply vendor-onboarding-checklist.md]: I like that idea, @dvvarela!  I do think that for VSTS and the AK Dev Env line items we'll need to be more specific than "RW" (read/write), but I like the matrix!

File: vendor-onboarding-checklist.md (hunk 9 above)
Comment (sztaylorakgov) [reply vendor-onboarding-checklist.md]: Yes, I think it's a good idea to have the forms.  If we can make time for it, I'd like to have a pre-built package of forms, or checklist pointing to the forms for each role.  That would make it so vendor folks know exactly what they need to complete, even if they do need some help.  I think some forms have default routing instructions on them.  So, we'll just need to be extra clear that the forms need to route through EISR PMO.

Comment (ctro) [reply vendor-onboarding-checklist.md]: For now we could just start with a simple list of the required documents

**Pull request merged**